### PR TITLE
feat: ask leader

### PR DESCRIPTION
**Pull request by fengjiachun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

DdlClient also needs to use ask_leader, refactor this part to make ask_leader easier to use

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/meta-client/src/client/ask_leader.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                    break;
+                }
+                Err(status) => {
+                    warn!("Failed to ask leader from: {addr}, {status}");
+                }
+            }
+        }
+
+        let leader = leader.context(error::NoLeaderSnafu)?;
+        if prev_leader.as_ref() != Some(&leader) {
```
Comment (WenyXu) [inline src/meta-client/src/client/ask_leader.rs:98]: Considers reading `prev_leader` again. The `leader` may be changed while calling the `ask_leader.'

File: src/meta-client/src/client/ask_leader.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            let mut leadership_group = self.leadership_group.write().unwrap();
+            leadership_group.leader = Some(leader.clone());
```
Comment (MichaelScofield) [inline src/meta-client/src/client/ask_leader.rs:100]: Better check again for `leadership_group.leader` and `leader` after write lock is acquired. Current value could be changed, double check is necessary.

**Comment by fengjiachun** [comment]
@MichaelScofield @

**Comment by fengjiachun** [comment]
@MichaelScofield @WenyXu PTAL

**Review by WenyXu — APPROVED** [review]

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**